### PR TITLE
chore: remove outdated updateSources script parts

### PR DESCRIPTION
**Pull request by web-padawan**
## Description

The script used to update sources when merging ITs contains a lot of outdated logic.
Removed some of it to keep it smaller. There might be still more things to remove.

## Type of change

- Internal change

### CODE REVIEW

File: scripts/updateSources.js [hunk 1]
```diff
@@ -77,14 +77,6 @@ async function main() {
   const textFieldVersionSource = 'vaadin-text-field-flow-parent/vaadin-text-field-flow/src/main/java/com/vaadin/flow/component/textfield/GeneratedVaadinTextField.java';
   const textFieldVersion = fs.readFileSync(textFieldVersionSource,'utf-8').split(/\n/).filter(l => l.startsWith('@NpmPackage'))[0];
 
-  await visitFilesRecursive(`${mod}/${id}-integration-tests/frontend`, (source, target, content) => {
-    if (/test-template.js$/.test(target)) {
-      target = target.replace('test-template', `${id}-test-template`);
-      content = content.replace('test-template', `${id}-test-template`);
-    }
-    return [target, content];
-  });
```
Comment (web-padawan) [inline scripts/updateSources.js:86]: Individual `test-template` parts are removed in #4216, only individual parts are now used.

File: scripts/updateSources.js [hunk 2]
```diff
@@ -93,8 +85,6 @@ async function main() {
   });
 
   await visitFilesRecursive(`${mod}/${id}-integration-tests/src`, (source, target, content) => {
-    // replace test-template localName for the new computed above
-    content = content.replace(/(\@Tag.*|\@JsModule.*|\$\(")test-template/g, `$1${id}-test-template`);
```
Comment (web-padawan) [inline scripts/updateSources.js:97]: Related to the above, the `test-template` files are now removed.

File: scripts/updateSources.js [hunk 3]
```diff
@@ -129,27 +119,6 @@ async function main() {
     content = content.replace(/\.attribute\("href", *"([^"]*)"\)/g, (...args) => {
       return `.attribute("href", "${wc}/${args[1]}")`;
     });
-    // Combo box - PreSelectedValueIT.selectedValueIsNotResetAfterClientResponse
-    // The test fails because when running in the project there is an iron-icon-set-svg element
```
Comment (web-padawan) [inline scripts/updateSources.js:133]: We no longer use `iron-iconset-svg` in V24 and the `IronIcon` was removed in #3934.
Furthermore, the actual ITs are already updated to use selectors specified below.

File: scripts/updateSources.js [hunk 4]
```diff
@@ -159,10 +128,6 @@ async function main() {
       return content;
     }
 
-    // Dialog: Workaround for https://github.com/vaadin/vaadin-confirm-dialog-flow/issues/136
-    // Since this project contains a dependency to vaadin-confirm-dialog, the height is different
-    // and the tests fail.
-    content = ignore_test_method(content, source, 'DialogIT.openAndCloseBasicDialog_labelRendered');
```
Comment (web-padawan) [inline scripts/updateSources.js:165]: The underlying issue was fixed, see https://github.com/vaadin/vaadin-confirm-dialog-flow/issues/136#issuecomment-687126222

File: scripts/updateSources.js [hunk 5]
```diff
@@ -175,7 +140,6 @@ async function main() {
     content = ignore_test_method(content, source, 'CustomGridIT.editorShouldHaveRightTitleWhenOpenedInNewItemMode');
     content = ignore_test_method(content, source, 'TreeGridPageSizeIT.treegridWithPageSize10_changeTo80_revertBackTo10');
     content = ignore_test_method(content, source, 'DynamicChangingChartIT.setConfiguration_changes_chart');
-    content = ignore_test_method(content, source, 'IronListIT.listWithComponentRendererWithBeansAndPlaceholder_scrollToBottom_placeholderIsShown');
```
Comment (web-padawan) [inline scripts/updateSources.js:178]: The `IronList` was removed completely in V24.

File: scripts/updateSources.js [hunk 6]
```diff
@@ -120,36 +110,10 @@ async function main() {
     if (/OverlayIT\.java$/.test(source)) {
       content = content.replace(/\/overlayselfattached/,`/${wc}$&`)
     }
-
-    // Accordion: Match textfield version
-    if (/AccordionInTemplate\.java$/.test(source)) {
-      content = content.replace(/@NpmPackage.*/,textFieldVersion);
-    }
```
Comment (web-padawan) [inline scripts/updateSources.js:127]: This fix is no longer relevant after Vaadin 20 where we aligned web components version.

File: scripts/updateSources.js [hunk 7]
```diff
@@ -216,8 +175,6 @@ async function main() {
     // charts v14
     content = content.replace('getTestView().getCanonicalName();', '"vaadin-charts/" + getTestView().getCanonicalName();');
 
-    content = content.replace('import com.vaadin.flow.demo.ComponentDemoTest','import com.vaadin.tests.ComponentDemoTest');
-    content = content.replace('import com.vaadin.flow.demo.TabbedComponentDemoTest','import com.vaadin.tests.TabbedComponentDemoTest');
```
Comment (web-padawan) [inline scripts/updateSources.js:220]: Both classes were coming from the demo-helpers module that has been removed in #2834.

File: scripts/updateSources.js [hunk 8]
```diff
@@ -216,11 +175,6 @@ async function main() {
...
-    content = content.replace('import com.vaadin.testbench.parallel.ParallelTest','import com.vaadin.tests.ParallelTest');
-    content = content.replace('import com.vaadin.flow.testutil.AbstractComponentIT','import com.vaadin.tests.AbstractComponentIT');
-    content = content.replace('import com.vaadin.flow.testutil.AbstractValidationTest','import com.vaadin.tests.AbstractValidationTest');
```
Comment (web-padawan) [inline scripts/updateSources.js:223]: These classes have been moved to `vaadin-flow-components-test-util` package.

File: scripts/updateSources.js [hunk 9]
```diff
@@ -159,27 +123,15 @@ async function main() {
...
 
-    // Dialog: Workaround for https://github.com/vaadin/vaadin-confirm-dialog-flow/issues/136
-    // Since this project contains a dependency to vaadin-confirm-dialog, the height is different
-    // and the tests fail.
-    content = ignore_test_method(content, source, 'DialogIT.openAndCloseBasicDialog_labelRendered');
     content = ignore_test_method(content, source, 'ServerSideEventsIT.chartClick_occured_eventIsFired');
-    content = ignore_test_method(content, source, 'GridDetailsRowIT.gridUpdateItemUpdateDetails');
-    content = ignore_test_method(content, source, 'BasicIT.customComboBox_circularReferencesInData_isEdited');
-    content = ignore_test_method(content, source, 'BasicIT.customComboBoxIsUsedForEditColumn');
-    content = ignore_test_method(content, source, 'BasicIT.checkboxEditorIsUsedForCheckboxColumn');
```
Comment (web-padawan) [inline scripts/updateSources.js:170]: These ITs have been re-enabled in #280, so there is no more need to ignore them here.

File: scripts/updateSources.js [hunk 10]
```diff
@@ -159,33 +123,14 @@ async function main() {
...
-
     content = ignore_test_method(content, source, 'EditOnClickIT.editButtonsAreHiddenIfEditOnClickIsEnabled');
     content = ignore_test_method(content, source, 'RendererIT.testRenderer_initialComponentRendererSet_rendersComponentsThatWork');
     content = ignore_test_method(content, source, 'RendererIT.testRenderer_componentRendererSet_rendersComponentsThatWork');
     content = ignore_test_method(content, source, 'CustomGridIT.editorShouldHaveRightTitleWhenOpenedInNewItemMode');
-    content = ignore_test_method(content, source, 'TreeGridPageSizeIT.treegridWithPageSize10_changeTo80_revertBackTo10');
-    content = ignore_test_method(content, source, 'DynamicChangingChartIT.setConfiguration_changes_chart');
```
Comment (web-padawan) [inline scripts/updateSources.js:177]: These tests were also re-enabled in #280.

File: scripts/updateSources.js [hunk 11]
```diff
@@ -159,33 +123,14 @@ async function main() {
...
-    content = ignore_test_method(content, source, 'IronListIT.listWithComponentRendererWithBeansAndPlaceholder_scrollToBottom_placeholderIsShown');
     content = ignore_test_method(content, source, 'VirtualListIT.listWithComponentRendererWithBeansAndPlaceholder_scrollToBottom_placeholderIsShown');
-    content = ignore_test_method(content, source, 'BasicChartIT.Chart_TitleCanBeChanged');
     content = ignore_test_method(content, source, 'MenuBarPageIT.disableItem_overflow_itemDisabled:262 NullPointer');
-    content = ignore_test_method(content, source, 'BasicIT.customEditorValueIsUpdatedByLeavingEditorWithTab');
     content = ignore_test_method(content, source, 'ValueChangeModeIT.test\\w+');
 
-    content = ignore_test_method(content, source, 'DynamicEditorKBNavigationIT.navigateBetweenEditorsUsingKeybaord');
-    content = ignore_test_method(content, source, 'IntegerFieldPageIT.integerOverflow_noException_valueSetToNull');
-    content = ignore_test_method(content, source, 'TreeGridHugeTreeNavigationIT.keyboard_navigation');
```
Comment (web-padawan) [inline scripts/updateSources.js:187]: These tests were also re-enabled in https://github.com/vaadin/flow-components/pull/280.

File: scripts/updateSources.js [hunk 12]
```diff
@@ -159,33 +123,12 @@ async function main() {
...
-    content = ignore_test_method(content, source, 'DialogIT.openAndCloseBasicDialog_labelRendered');
-    content = ignore_test_method(content, source, 'ServerSideEventsIT.chartClick_occured_eventIsFired');
-    content = ignore_test_method(content, source, 'GridDetailsRowIT.gridUpdateItemUpdateDetails');
-    content = ignore_test_method(content, source, 'BasicIT.customComboBox_circularReferencesInData_isEdited');
-    content = ignore_test_method(content, source, 'BasicIT.customComboBoxIsUsedForEditColumn');
-    content = ignore_test_method(content, source, 'BasicIT.checkboxEditorIsUsedForCheckboxColumn');
-
     content = ignore_test_method(content, source, 'EditOnClickIT.editButtonsAreHiddenIfEditOnClickIsEnabled');
-    content = ignore_test_method(content, source, 'RendererIT.testRenderer_initialComponentRendererSet_rendersComponentsThatWork');
-    content = ignore_test_method(content, source, 'RendererIT.testRenderer_componentRendererSet_rendersComponentsThatWork');
```
Comment (web-padawan) [inline scripts/updateSources.js:174]: These tests were also re-enabled in https://github.com/vaadin/flow-components/pull/280.

File: scripts/updateSources.js [hunk 13]
```diff
@@ -159,32 +123,9 @@ async function main() {
...
     content = ignore_test_method(content, source, 'CustomGridIT.editorShouldHaveRightTitleWhenOpenedInNewItemMode');
-    content = ignore_test_method(content, source, 'TreeGridPageSizeIT.treegridWithPageSize10_changeTo80_revertBackTo10');
-    content = ignore_test_method(content, source, 'DynamicChangingChartIT.setConfiguration_changes_chart');
-    content = ignore_test_method(content, source, 'IronListIT.listWithComponentRendererWithBeansAndPlaceholder_scrollToBottom_placeholderIsShown');
     content = ignore_test_method(content, source, 'VirtualListIT.listWithComponentRendererWithBeansAndPlaceholder_scrollToBottom_placeholderIsShown');
-    content = ignore_test_method(content, source, 'BasicChartIT.Chart_TitleCanBeChanged');
-    content = ignore_test_method(content, source, 'MenuBarPageIT.disableItem_overflow_itemDisabled:262 NullPointer');
-    content = ignore_test_method(content, source, 'BasicIT.customEditorValueIsUpdatedByLeavingEditorWithTab');
-    content = ignore_test_method(content, source, 'ValueChangeModeIT.test\\w+');
```
Comment (web-padawan) [inline scripts/updateSources.js:183]: All the tests in `ValueChangeModeIT` were also re-enabled in #280.

File: scripts/updateSources.js [hunk 14]
```diff
@@ -216,22 +156,9 @@ async function main() {
...
 
-    content = content.replace('import com.vaadin.flow.demo.ComponentDemoTest','import com.vaadin.tests.ComponentDemoTest');
-    content = content.replace('import com.vaadin.flow.demo.TabbedComponentDemoTest','import com.vaadin.tests.TabbedComponentDemoTest');
-    content = content.replace('import com.vaadin.testbench.parallel.ParallelTest','import com.vaadin.tests.ParallelTest');
-    content = content.replace('import com.vaadin.flow.testutil.AbstractComponentIT','import com.vaadin.tests.AbstractComponentIT');
-    content = content.replace('import com.vaadin.flow.testutil.AbstractValidationTest','import com.vaadin.tests.AbstractValidationTest');
     content = content.replace('import com.vaadin.testbench.annotations.BrowserConfiguration;','');
     content = content.replace(/.*@BrowserConfiguration.*/,'');
 
-    // Remove W3C workaround from Grid tests.
```
Comment (web-padawan) [inline scripts/updateSources.js:227]: The original workaround was removed in https://github.com/vaadin/vaadin-grid-flow/pull/1101.

File: scripts/updateSources.js [hunk 15]
```diff
@@ -77,14 +77,6 @@ async function main() {
   const textFieldVersionSource = 'vaadin-text-field-flow-parent/vaadin-text-field-flow/src/main/java/com/vaadin/flow/component/textfield/GeneratedVaadinTextField.java';
   const textFieldVersion = fs.readFileSync(textFieldVersionSource,'utf-8').split(/\n/).filter(l => l.startsWith('@NpmPackage'))[0];
```
Comment (sissbruecker) [inline scripts/updateSources.js:78]: These are unused now.

**Review by sissbruecker — APPROVED** [review]

File: scripts/updateSources.js (hunk 15 above)
Comment (web-padawan) [reply scripts/updateSources.js:78]: Thanks, removed.

**Pull request merged**